sbi/inference/posterior.py

```diff
@@ -129,12 +122,89 @@ def set_default_x(self, x: Tensor) -> "NeuralPosterior":
         Returns:
             `NeuralPosterior` that will use a default `x` when not explicitly passed.
         """
-        processed_x = process_x(x, self._x_shape)
-        self._warn_if_posterior_was_focused_on_different_x(processed_x)
-        self._x = processed_x
 
         return self
 
+    @property
+    def mcmc_method(self) -> str:
+        """Returns MCMC method."""
+        return self._mcmc_method
+
+    @mcmc_method.setter
+    def mcmc_method(self, method: str) -> None:
+        """See `set_mcmc_method`."""
+        self.set_mcmc_method(method)
+
+    def set_mcmc_method(self, method: str) -> "NeuralPosterior":
```

Comment:
If a `set` function also returns, I would suggest change the name to e.g. `with_mcmc_method(method)` or similar.  Also important to be very clear about returns in the docstring ("Set ... and return... ") .